subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -46,6 +46,26 @@ Forward compatibility was retained, but the following exports might cause a larg
 
 The `link:{javadocPath}/javadoc/org/gradle/api/attributes/AttributesSchema.html#setAttributeDisambiguationPrecedence(List)--[AttributeSchema.setAttributeDisambiguationPrecedence(List)]` and `link:{javadocPath}/javadoc/org/gradle/api/attributes/AttributesSchema.html#getAttributeDisambiguationPrecedence()--[AttributeSchema.getAttributeDisambiguationPrecedence()]` methods now accept and return `List` instead of `Collection` to better indicate that the order of the elements in those collection is significant.
 
+[[strict-kotlin-dsl-precompiled-scripts-accessors]]
+=== Strict Kotlin DSL precompiled script plugins accessors generation
+
+Type safe Kotlin DSL accessors generation for precompiled script plugins does not fail the build by default if a plugin requested in such precompiled scripts fails to be applied.
+Because the cause could be environmental and for backwards compatibility reasons, this behaviour hasn't changed yet.
+
+Back in Gradle 7.1 the `:generatePrecompiledScriptPluginAccessors` task responsible for the accessors generation has been marked as non cacheable by default.
+The `org.gradle.kotlin.dsl.precompiled.accessors.strict` system property was introduced in order to offer an opt-in to a stricter mode of operation that fails the build when a plugin application fails, and enable the build cache for that task.
```

Comment:
```suggestion
The `org.gradle.kotlin.dsl.precompiled.accessors.strict` system property was introduced in order to offer an opt in to a stricter mode of operation that fails the build when a plugin application fails, and enable the build cache for that task.
```

Reply:
IntelliJ was telling me that the noun needs the `-` but the verb does not.